ignite/services/plugin/internal/execute.go

```diff
@@ -0,0 +1,21 @@
+package plugininternal
+
+import (
+	"context"
+
+	pluginsconfig "github.com/ignite/cli/v28/ignite/config/plugins"
+	"github.com/ignite/cli/v28/ignite/services/plugin"
+)
+
+// Execute starts and executes a plugin, then shutdowns it.
+func Execute(ctx context.Context, path string, args ...string) error {
+	plugins, err := plugin.Load(ctx, []pluginsconfig.Plugin{{Path: path}})
+	if err != nil {
+		return err
+	}
+	defer plugins[0].KillClient()
+	if plugins[0].Error != nil {
+		return plugins[0].Error
+	}
+	return plugins[0].Interface.Execute(ctx, &plugin.ExecutedCommand{Args: args}, plugin.NewClientAPI())
```

Comment:
It would be also possible here to send the chain as an option to the ClientAPI I believe, something in the lines of:

```suggestion
	c := chain.New(path)
	return plugins[0].Interface.Execute(ctx, &plugin.ExecutedCommand{Args: args}, plugin.NewClientAPI(plugin.WithChain(c)))
```

Reply:
I see your point, but `path` here is not the path to the chain app, it's the path to the plugin.
I think the only way to pass a `*chain.Chain` here is to add `plugin.APIOptions` to the `Execute` function, wdyt?

Reply:
> I think the only way to pass a `*chain.Chain` here is to add `plugin.APIOptions` to the `Execute` function, wdyt?

Makes sense 👍

That would enable us to keep "parity" with the current plugins implementation by sending any additional public plugin options to the private `Execute()`, otherwise we wouldn't potentially be able to execute some existing plugins from this internal implementation.

Maybe the options could be only available to the `Execute()` here, so "args" would be another option:
```go
func WithArgs(args ...string) Option {}
func WithChain(*chain.Chain) Option {}

func Execute(ctx context.Context, path string, options ...Options) error {}
```

Reply:
I'm going to add the `APIOptions` to `Execute`, but I'm not going to add another option type because I think Ignite overuses the option pattern. Having a function that takes options and calls another function that takes a different type of options just makes things too hard to read.

Reply:
Let's wait for my change before resolving the discussion ty :pray:

Reply:
So here is my change [8e8cddc](https://github.com/ignite/cli/pull/3899/commits/8e8cddc3550517112d3996ee59902bc8c45d6a8d)